### ISSUE TITLE
Crash on `small_generating_set`

### ISSUE THREAD

**Issue by HechtiDerLachs** [oscar-system/Oscar.jl#3499]
I was trying to fix the first test in `test/AlgebraicGeometry/schemes/K3.jl` on #3490 . But it seems that `small_generating_set` is giving me a hard time. To reproduce, try the following:
```

  IP1 = projective_space(GF(29), ["s", "t"])

  O0 = twisting_sheaf(IP1, 0)
  O4 = twisting_sheaf(IP1, -4)
  O6 = twisting_sheaf(IP1, -6)

  E = direct_sum([O0, O4, O6])

  X_proj = projectivization(E, var_names=["z", "x", "y"])

  X = covered_scheme(X_proj)

  U = affine_charts(X)[1]
  (x, y, t) = gens(OO(U))
  ft = y^2 - (x^3 + 21*x + (28*t^7+18))
  I = IdealSheaf(X, U, [ft])

  # The fabulous K3-surface. Almost.
  inc_S = Oscar.CoveredClosedEmbedding(X, I)
  @test I === image_ideal(inc_S)
  S = domain(inc_S)

  I_sing = Oscar.ideal_sheaf_of_singular_locus(S)
  I_sing_X = small_generating_set(pushforward(inc_S)(I_sing))
             # happens to be radical in this example -- no radical needed here
  @test scheme(I_sing) === S
  @test scheme(I_sing_X) === X

  prX = blow_up(I_sing_X)
  E1 = Oscar.exceptional_divisor(prX)
  X1 = domain(prX)
  Y1, inc_Y1, pr_Y1 = strict_transform(prX, inc_S)


  I_sing_Y1 = Oscar.ideal_sheaf_of_singular_locus(Y1)
  I_sing_X1 = small_generating_set(pushforward(inc_Y1)(I_sing_Y1))
             # happens to be radical in this example -- no radical needed here
  prX2 = blow_up(I_sing_X1, covering=Oscar.simplified_covering(X1),
                var_name="t")
  E2 = exceptional_divisor(prX2)
  X2 = domain(prX2)
  Y2, inc_Y2, pr_Y2 = strict_transform(prX2, inc_Y1)
  simplify!(Y2)
  # @show has_attribute(Y2, :simplified_covering)
  I_sing_Y2 = Oscar.ideal_sheaf_of_singular_locus(Y2)
  I_sing_X2 = small_generating_set(pushforward(inc_Y2)(I_sing_Y2))
             # happens to be radical in this example -- no radical needed here
  prX3 = blow_up(I_sing_X2, covering=Oscar.simplified_covering(X2),
                var_name="u")
  E3 = exceptional_divisor(prX3)
  X3 = domain(prX3)
  Y3, inc_Y3, pr_Y3 = strict_transform(prX3, inc_Y2)
  simplify!(Y3)
  # @show has_attribute(Y3, :simplified_covering)
  I_sing_Y3 = Oscar.ideal_sheaf_of_singular_locus(Y3)
  I_sing_X3 = pushforward(inc_Y3)(I_sing_Y3)

  # Now the singular locus consists of two points. 
  # We blow them up successively rather than at once. 
  #println("starting primary decomposition of singular locus")
  decomp = Oscar.maximal_associated_points(I_sing_X3)
  a = first(decomp)
  U = affine_charts(scheme(a))
  [parent(first(small_generating_set(a(V)))) === OO(V) for V in U]
```
I get a full crash of julia with the following stacktrace:
```
julia> [parent(first(small_generating_set(a(V)))) === OO(V) for V in U]

[4329] signal (11.1): Segmentation fault
in expression starting at REPL[46]:1
_Z6initSLP10sip_sidealS0_P10skStrategy at /home/matthias/.julia/artifacts/e92b64faf2c02c0cdfb5b186449a8ecfcd29a990/lib/libSingular.so (unknown line)
_Z12initBuchMoraP10sip_sidealS0_P10skStrategy at /home/matthias/.julia/artifacts/e92b64faf2c02c0cdfb5b186449a8ecfcd29a990/lib/libSingular.so (unknown line)
_Z3bbaP10sip_sidealS0_P6intvecS2_P10skStrategy at /home/matthias/.julia/artifacts/e92b64faf2c02c0cdfb5b186449a8ecfcd29a990/lib/libSingular.so (unknown line)
_Z8kMin_stdP10sip_sidealS0_6tHomogPP6intvecRS0_S3_ii at /home/matthias/.julia/artifacts/e92b64faf2c02c0cdfb5b186449a8ecfcd29a990/lib/libSingular.so (unknown line)
_Z16id_MinStd_helperP10sip_sidealP8ip_sringb at /home/matthias/.julia/artifacts/fbcc30fd8943f6c33a70746db41f36008fbee57a/lib/libsingular_julia.so (unknown line)
_ZNSt17_Function_handlerIFSt5tupleIJP10sip_sidealS2_EES2_P8ip_sringbEPS6_E9_M_invokeERKSt9_Any_dataOS2_OS5_Ob at /home/matthias/.julia/artifacts/fbcc30fd8943f6c33a70746db41f36008fbee57a/lib/libsingular_julia.so (unknown line)
_ZN5jlcxx6detail11CallFunctorISt5tupleIJP10sip_sidealS4_EEJS4_P8ip_sringbEE5applyEPKvNS_13WrappedCppPtrESB_b at /home/matthias/.julia/artifacts/fbcc30fd8943f6c33a70746db41f36008fbee57a/lib/libsingular_julia.so (unknown line)
id_MinStd at /home/matthias/.julia/packages/CxxWrap/5IZvn/src/CxxWrap.jl:624
#mstd#150 at /home/matthias/.julia/packages/Singular/N0xXX/src/ideal/ideal.jl:687 [inlined]
mstd at /home/matthias/.julia/packages/Singular/N0xXX/src/ideal/ideal.jl:685
unknown function (ip: 0x7fa1683cfe32)
_jl_invoke at /cache/build/builder-amdci4-6/julialang/julia-release-1-dot-10/src/gf.c:2894 [inlined]
ijl_apply_generic at /cache/build/builder-amdci4-6/julialang/julia-release-1-dot-10/src/gf.c:3076
small_generating_set at /home/matthias/Kaiserslautern/neuer_Oscar_Klon/Oscar.jl/src/Rings/MPolyQuo.jl:1796
unknown function (ip: 0x7fa105e39195)
_jl_invoke at /cache/build/builder-amdci4-6/julialang/julia-release-1-dot-10/src/gf.c:2894 [inlined]
ijl_apply_generic at /cache/build/builder-amdci4-6/julialang/julia-release-1-dot-10/src/gf.c:3076
#3 at ./none:0
unknown function (ip: 0x7fa105e25845)
_jl_invoke at /cache/build/builder-amdci4-6/julialang/julia-release-1-dot-10/src/gf.c:2894 [inlined]
ijl_apply_generic at /cache/build/builder-amdci4-6/julialang/julia-release-1-dot-10/src/gf.c:3076
iterate at ./generator.jl:47 [inlined]
collect_to! at ./array.jl:892
collect_to_with_first! at ./array.jl:870 [inlined]
collect at ./array.jl:844
unknown function (ip: 0x7fa105e22d35)
_jl_invoke at /cache/build/builder-amdci4-6/julialang/julia-release-1-dot-10/src/gf.c:2894 [inlined]
ijl_apply_generic at /cache/build/builder-amdci4-6/julialang/julia-release-1-dot-10/src/gf.c:3076
jl_apply at /cache/build/builder-amdci4-6/julialang/julia-release-1-dot-10/src/julia.h:1982 [inlined]
do_call at /cache/build/builder-amdci4-6/julialang/julia-release-1-dot-10/src/interpreter.c:126
eval_value at /cache/build/builder-amdci4-6/julialang/julia-release-1-dot-10/src/interpreter.c:223
eval_stmt_value at /cache/build/builder-amdci4-6/julialang/julia-release-1-dot-10/src/interpreter.c:174 [inlined]
eval_body at /cache/build/builder-amdci4-6/julialang/julia-release-1-dot-10/src/interpreter.c:617
jl_interpret_toplevel_thunk at /cache/build/builder-amdci4-6/julialang/julia-release-1-dot-10/src/interpreter.c:775
jl_toplevel_eval_flex at /cache/build/builder-amdci4-6/julialang/julia-release-1-dot-10/src/toplevel.c:934
jl_toplevel_eval_flex at /cache/build/builder-amdci4-6/julialang/julia-release-1-dot-10/src/toplevel.c:877
jl_toplevel_eval_flex at /cache/build/builder-amdci4-6/julialang/julia-release-1-dot-10/src/toplevel.c:877
ijl_toplevel_eval_in at /cache/build/builder-amdci4-6/julialang/julia-release-1-dot-10/src/toplevel.c:985
eval at ./boot.jl:385 [inlined]
eval_user_input at /cache/build/builder-amdci4-6/julialang/julia-release-1-dot-10/usr/share/julia/stdlib/v1.10/REPL/src/REPL.jl:150
repl_backend_loop at /cache/build/builder-amdci4-6/julialang/julia-release-1-dot-10/usr/share/julia/stdlib/v1.10/REPL/src/REPL.jl:246
#start_repl_backend#46 at /cache/build/builder-amdci4-6/julialang/julia-release-1-dot-10/usr/share/julia/stdlib/v1.10/REPL/src/REPL.jl:231
start_repl_backend at /cache/build/builder-amdci4-6/julialang/julia-release-1-dot-10/usr/share/julia/stdlib/v1.10/REPL/src/REPL.jl:228
_jl_invoke at /cache/build/builder-amdci4-6/julialang/julia-release-1-dot-10/src/gf.c:2894 [inlined]
ijl_apply_generic at /cache/build/builder-amdci4-6/julialang/julia-release-1-dot-10/src/gf.c:3076
#run_repl#59 at /cache/build/builder-amdci4-6/julialang/julia-release-1-dot-10/usr/share/julia/stdlib/v1.10/REPL/src/REPL.jl:389
run_repl at /cache/build/builder-amdci4-6/julialang/julia-release-1-dot-10/usr/share/julia/stdlib/v1.10/REPL/src/REPL.jl:375
jfptr_run_repl_91689.1 at /home/matthias/Kaiserslautern/Julia/julia-1.10.0/lib/julia/sys.so (unknown line)
_jl_invoke at /cache/build/builder-amdci4-6/julialang/julia-release-1-dot-10/src/gf.c:2894 [inlined]
ijl_apply_generic at /cache/build/builder-amdci4-6/julialang/julia-release-1-dot-10/src/gf.c:3076
#1013 at ./client.jl:432
jfptr_YY.1013_82677.1 at /home/matthias/Kaiserslautern/Julia/julia-1.10.0/lib/julia/sys.so (unknown line)
_jl_invoke at /cache/build/builder-amdci4-6/julialang/julia-release-1-dot-10/src/gf.c:2894 [inlined]
ijl_apply_generic at /cache/build/builder-amdci4-6/julialang/julia-release-1-dot-10/src/gf.c:3076
jl_apply at /cache/build/builder-amdci4-6/julialang/julia-release-1-dot-10/src/julia.h:1982 [inlined]
jl_f__call_latest at /cache/build/builder-amdci4-6/julialang/julia-release-1-dot-10/src/builtins.c:812
#invokelatest#2 at ./essentials.jl:887 [inlined]
invokelatest at ./essentials.jl:884 [inlined]
run_main_repl at ./client.jl:416
exec_options at ./client.jl:333
_start at ./client.jl:552
jfptr__start_82703.1 at /home/matthias/Kaiserslautern/Julia/julia-1.10.0/lib/julia/sys.so (unknown line)
_jl_invoke at /cache/build/builder-amdci4-6/julialang/julia-release-1-dot-10/src/gf.c:2894 [inlined]
ijl_apply_generic at /cache/build/builder-amdci4-6/julialang/julia-release-1-dot-10/src/gf.c:3076
jl_apply at /cache/build/builder-amdci4-6/julialang/julia-release-1-dot-10/src/julia.h:1982 [inlined]
true_main at /cache/build/builder-amdci4-6/julialang/julia-release-1-dot-10/src/jlapi.c:582
jl_repl_entrypoint at /cache/build/builder-amdci4-6/julialang/julia-release-1-dot-10/src/jlapi.c:731
main at julia (unknown line)
__libc_start_main at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
unknown function (ip: 0x4010b8)
Allocations: 159104781 (Pool: 158966494; Big: 138287); GC: 43
Segmentation fault (core dumped)
```
Any ideas what's going on? 

CC: @hannes14

**Comment by joschmitt** [comment]
Here is a minimal "non-working" example that produces the error:
```
julia> F = GF(29);

julia> R, (x, y, z) = F["x", "y", "z"];

julia> Q, _ = quo(R, ideal(R, [0, 0, 0]));

julia> I = ideal(Q, [one(Q)])
ideal(1)

julia> small_generating_set(I)
[681153] signal (11.1): Speicherzugriffsfehler
...
```
(Note to self: the printing of ideals in quotient rings still needs some love.)

**Comment by afkafkafk13** [comment]
This is `Singular.mstd`crashing for the input
singular_generators(I.gens) = Singular ideal over Singular polynomial quotient ring (ZZ/29),(x,y,z),(dp(3),C) with generators (1)
Here the Singular qring needs to be avoided. I will provide a fix for this.

**Comment by HechtiDerLachs** [comment]
> Here the Singular qring needs to be avoided. 

Why is avoidance the right approach to "fix" this issue? The input is sane, isn't it? 

Edit: I already provide a temporary workaround in #3490, referring to this issue . So doing another one might be unnecessary work at this point.

**Comment by hannes14** [comment]
fixed in Singular: it happened with ALL operations with Groebner bases over quotient ring wrt. an ideal with 0 generators.
Need to create a new Singular_jll etc.

**Comment by afkafkafk13** [comment]
@hannes14 Thank you for the quick response.

**Comment by fingolfin** [comment]
We'll make a new Singular.jl release later today.

**Comment by fingolfin** [comment]
Fixed in Singular 0.22.6
